dali/pipeline/data/tensor_list.cc

```diff
@@ -1103,14 +1103,19 @@ void TensorList<Backend>::UpdatePropertiesFromSamples(bool contiguous) {
   // assume that the curr_num_tensors_ is valid
   DALI_ENFORCE(curr_num_tensors_ > 0,
                "Unexpected empty output of per-sample operator. Internal DALI error.");
-  type_ = tensors_[0].type_info();
-  sample_dim_ = tensors_[0].shape().sample_dim();
-  shape_.resize(curr_num_tensors_, sample_dim_);
-  layout_ = tensors_[0].GetMeta().GetLayout();
-  pinned_ = tensors_[0].is_pinned();
-  order_ = tensors_[0].order();
-  device_ = tensors_[0].device_id();
-  contiguous_buffer_.set_order(order_);
+  for (int i = 0; i < curr_num_tensors_; i++) {
+    // if tensor is empty it can be uninitialized so we don't really care about its properties
```

Comment:
What if we detected the first non-empty one, and propagated the setup to the empty ones?